django/db/models/lookups.py

```diff
@@ -361,7 +361,12 @@ def process_rhs(self, compiler, connection):
                 rhs = self.rhs
 
             if not rhs:
-                raise EmptyResultSet
+                if self.__class__.__name__ == "NotIn":
```

Comment:
Instead of putting this in `In`, I think it's better to override `process_rhs` in `NotIn` itself. Perhaps you could use `try` and call `super().process_lhs`, then `except EmptyResultSet`.